services/provider/client/client.go

```diff
@@ -10,82 +10,103 @@ import (
 )
 
 type OCSProviderClient struct {
-	Client  pb.OCSProviderClient
-	timeout time.Duration
+	Client     pb.OCSProviderClient
+	clientConn *grpc.ClientConn
+	timeout    time.Duration
 }
 
-// NewProviderClient creates a client to talk to OCS provider server
-func NewProviderClient(cc *grpc.ClientConn, timeout time.Duration) *OCSProviderClient {
-	return &OCSProviderClient{Client: pb.NewOCSProviderClient(cc), timeout: timeout}
-}
-
-// NewGRPCConnection returns a grpc client connection which can be used to create the consumer client
-// Note: Close the connection after use
-func NewGRPCConnection(serverAddr string, timeout time.Duration) (*grpc.ClientConn, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+// NewProviderClient creates a client to talk to the external OCS storage provider server
+func NewProviderClient(ctx context.Context, serverAddr string, timeout time.Duration) (*OCSProviderClient, error) {
+	apiCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 
 	opts := []grpc.DialOption{
 		grpc.WithInsecure(),
 		grpc.WithBlock(),
 	}
 
-	conn, err := grpc.DialContext(ctx, serverAddr, opts...)
+	conn, err := grpc.DialContext(apiCtx, serverAddr, opts...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to dial: %v", err)
 	}
 
-	return conn, err
+	return &OCSProviderClient{
+		Client:     pb.NewOCSProviderClient(conn),
+		clientConn: conn,
+		timeout:    timeout}, nil
+}
+
+// Close closes the gRPC connection of the external OCS storage provider client
+func (cc *OCSProviderClient) Close() {
```

Comment:
The OCSProviderClient (for better or worse) is an abstraction on top of the underlying GRPC code. c
close() is to prevent leakage of the GRPC implementation outside the abstraction even if it was a single line of code and not three

Reply:
Ah, okay.